src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -649,14 +649,37 @@ public static IEnumerable<String> FilterCompatibleSources(Architecture chosenPla
         }
 
         /// <summary>
-        /// Returns true if source settings are incomaptible with target settings.
+        /// Returns true if target framework is incompatible with any source framework, that is run needs to be aborted
+        /// </summary>
+        public static bool IsFrameworkIncompatible(Framework chosenFramework, IDictionary<String, Framework> sourceFrameworks)
```

Comment:
>public static bool IsFrameworkIncompatibl [](start = 8, length = 41)

do we actually need this.

Reply:
We need this to determine if the frameworks(full vs core scenario is happening), hence needing to be aborted. Restructured the other `IsFrameworkIncompatible`